src/pool-bin/BinPoolManager.sol

```diff
@@ -187,20 +187,19 @@ contract BinPoolManager is IBinPoolManager, Fees, Extsload {
         PoolId id = key.toId();
         _checkPoolInitialized(id);
 
-        uint24 totalSwapFee;
-        if (key.fee.isDynamicSwapFee()) {
-            totalSwapFee = IBinDynamicFeeManager(address(key.hooks)).getFeeForSwapInSwapOut(
+        uint24 lpFee;
+        if (key.fee.isDynamicLPFee()) {
+            lpFee = IBinDynamicFeeManager(address(key.hooks)).getFeeForSwapInSwapOut(
                 msg.sender, key, swapForY, 0, amountOut
             );
-            if (totalSwapFee > SwapFeeLibrary.TEN_PERCENT_FEE) revert FeeTooLarge();
         } else {
             // clear the top 4 bits since they may be flagged
-            totalSwapFee = key.fee.getInitialSwapFee();
+            lpFee = key.fee.getInitialLPFee();
         }
+        lpFee.validate(LPFeeLibrary.TEN_PERCENT_FEE);
 
         (amountIn, amountOutLeft, fee) = pools[id].getSwapIn(
-            BinPool.SwapViewParams({swapForY: swapForY, binStep: key.parameters.getBinStep(), fee: totalSwapFee}),
-            amountOut
+            BinPool.SwapViewParams({swapForY: swapForY, binStep: key.parameters.getBinStep(), lpFee: lpFee}), amountOut
```

Comment:
For bin-pool, the fee change involves 4 functions:
1. getSwapIn
2. getSwapOut
3. Swap
4. Mint